### PR TITLE
Make Living::addEffect() return bool

### PR DESCRIPTION
**Pull request by Muqsit**
## Introduction

This will let you know if an effect has been successfully applied or not.

### API changes

Living::addEffect() returns whether the effect has successfully been applied or not.

## Tests

```php
/** @var Living $living */
$effect = Effect::getEffect(Effect::SPEED)->setDuration(INT32_MAX)->setAmplifier(1);
var_dump($living->addEffect($effect));//bool(true)

$effect->setAmplifier(0);
var_dump($living->addEffect($effect));//bool(false)
```

### CODE REVIEW

File: src/pocketmine/entity/Living.php [hunk 1]
```diff
@@ -231,17 +231,19 @@ public function hasEffect(int $effectId) : bool{
 	 * Adds an effect to the mob.
 	 * If a weaker effect of the same type is already applied, it will be replaced.
 	 * If a weaker or equal-strength effect is already applied but has a shorter duration, it will be replaced.
+	 * Returns whether the effect has been successfully applied.
```
Comment (SOF3) [inline src/pocketmine/entity/Living.php]: This sentence fits better on line 237 behind `@return bool`.

**Comment by dktapps** [comment]
that change would not be within the scope of this pull request.

**Comment by Muqsit** [comment]
with a change in the title and description

**Comment by dktapps** [comment]
that would be combining two unrelated changes into the same PR, which goes against the guidelines.

**Comment by Muqsit** [comment]
you win bruv 👍

File: src/pocketmine/entity/Living.php [hunk 2]
```diff
@@ -233,15 +233,18 @@ public function hasEffect(int $effectId) : bool{
 	 * If a weaker or equal-strength effect is already applied but has a shorter duration, it will be replaced.
 	 *
 	 * @param Effect $effect
+	 *
+ 	 * Returns whether the effect has been successfully applied.
```
Comment (dktapps) [inline src/pocketmine/entity/Living.php]: I think @SOF3 meant something more like `@return bool Whether the effect has been successfully applied`

Comment (Muqsit) [reply src/pocketmine/entity/Living.php]: ah yeah he probably did

**Pull request merged**